src/itensor.jl

```diff
@@ -1001,6 +1001,25 @@ permute(T::ITensor,
         inds::Index...) = permute(T,
                                   IndexSet(inds...))
 
+"""
+    permute!(T::ITensors, inds)
+    permute!(T::ITensors, inds::Index...)
+
+Modify the ITensor T to have indices permuted according
+to the input indices inds. The storage of the ITensor
+is permuted accordingly.
+"""
+function Base.permute!(T::ITensor{N},new_inds) where {N}
+  Tp = permute(T,new_inds)
+  T.store = Tp.store
```

Comment:
Could you rewrite this as `setstore!(T, store(Tp))`? Generally I prefer using functions to access/modify fields since it is a bit more robust.

Reply:
Yes, good point!